design/aws-ecs-priv-registry-support.md

```diff
@@ -75,56 +69,39 @@ export interface IContainerImage {
 ```
 
 
-## Part 2: Extend `DockerHubImage` class to include optional creds
+# Part 2: Rename `DockerHubImage` class to `WebHostedImage`, add optional creds
 
-The `DockerHubImage` construct, which currently takes a string for image name, can be extended to take in an (optional) "repositoryCredentials" field of type `IRepositoryCreds`:
+The `DockerHubImage` construct will be renamed to `WebHostedImage`, and augmented to take in optional "credentials" via keyword props:
 ```ts
-export class DockerHubImage implements IContainerImage {
-  // add repositoryCredentials to constructor
-  constructor(public readonly imageName: string, public readonly repositoryCredentials: RepositoryCreds) {
-  }
-
-  public bind(_containerDefinition: ContainerDefinition): void {
-    // Nothing to do
-  }
+// define props
+export interface WebHostedImageProps {
+    credentials: RepositoryCreds;
 }
-```
-
-Example use:
-```ts
-const taskDefinition = new ecs.Ec2TaskDefinition(stack, 'TaskDef');
 
-const secret = secretsManager.Secret.import(stack, 'myRepoSecret', {
-  secretArn: 'arn:aws:secretsmanager:.....'
-})
-
-taskDefinition.AddContainer('myPrivateContainer', {
-  image: ecs.ContainerImage.fromDockerHub('userx/test', ecs.RepositoryCreds.fromSecret(secret));
-});
+// "DockerHubImage" class --> "WebHostedImage"
+export class WebHostedImage implements IContainerImage {
+  public readonly imageName: string;
+  public readonly credentials: IRepositoryCreds;
 
-```
-
-
-## Part 3: Add `fromPrivateRepository` method on `ContainerImage`, return new `PrivateImage` subclass
-
-For non-DockerHub privately hosted images, we can add a new subclass:
-```ts
-export class PrivateImage implements IContainerImage {
-    constructor(public readonly image: string, public readonly repositoryCredentials: RepositoryCreds) {
+  // add credentials to constructor
+  constructor(imageName: string, props: WebHostedImageProps) {
```

Comment:
Since all properties will be optional, default to `props: WebHostedImageProps = {}`.